envoy-control-core/src/main/resources/lua/ingress_handler.lua

```diff
@@ -0,0 +1,34 @@
+
+function envoy_on_request(handle)
+    local path = handle:headers():get(":path")
+    local method = handle:headers():get(":method")
+    local service_name = handle:headers():get("x-service-name")
+    local xff_header = handle:headers():get("x-forwarded-for")
+    local metadata = handle:streamInfo():dynamicMetadata()
+    metadata:set("envoy.filters.http.lua", "request.info.path", path)
+    metadata:set("envoy.filters.http.lua", "request.info.method", method)
+    metadata:set("envoy.filters.http.lua", "request.info.service_name", service_name)
+    metadata:set("envoy.filters.http.lua", "request.info.xff_header", xff_header)
+end
+
+function envoy_on_response(handle)
+    local rbacMetadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.rbac")
+
+    if rbacMetadata == nil or rbacMetadata["shadow_engine_result"] ~= "denied" then
+        return
```

Comment:
Does this branch mean we have an error with our setup? If incoming permissions are enabled then shadow rules should be enabled right? Should we log this or add some metric?

Reply:
* `rbacMetadata["shadow_engine_result"] ~= "denied"` - this is a normal situation, when request is allowed by RBAC
* `rbacMetadata == nil` - it is here just to be safe